module-2/backend/week_two.md

```diff
@@ -5,37 +5,201 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+</br>
+</br>
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
+</br>
+</br>
+
+* Db:create - this creates a database if it doesn't already exist
+* Db:create_migration - this preps the migration file so we can indicate the columns/attributes that should be in the table.
+* Db:migrate - this creates or updates tables in our database
+</br>
+</br>
 3. What's the difference between agile workflow and waterfall method?
+</br>
+</br>
+
+* Agile - Agile workflow uses many iterations where we start by determining one problem, writing code, testing, and analysis before we begin working on a new problem/feature. 
+* Waterfall - This methodology calls for fully planning upfront. All decisions about the project/code are made before any code is written or tested. There are no iterations. 
+
+</br>
+</br>
 4. What can you expect from a group as you begin working together? As you continue working together?
+</br>
+</br>
+
+In the beginning when groups are just getting to know each other, they are often very polite/pleasant and just kind of feeling each other out. Over time, as members become more comfortable, more ideas will come up as well as sometimes conflict and tension as the group learns how to work together better.
+
+</br>
+</br>
 5. What two columns does `t.timestamps null: false` create in our database?
+</br>
+</br>
+
+* `Created_at`
+* `Updated_at`
+</br>
+</br>
 6. What cURL flag can you use to send a `POST` request?
+</br>
+</br>
 7. What case does JSON (and JavaScript) use for multi-word variables?
+</br>
+camel
+</br>
 8. What case does Ruby use for multi-word variables?
+</br>
+snake
+</br>
 9. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+</br>
+
+* A school `has_many` teachers and a teacher `belongs_to` a school.
+
+</br>
 10. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
 11. Give an example of when you might want to store information besides ids on a join table.
+</br>
+
+* You might do this if data on the join table is distinct from any of the info stored on the tables it joins. For example on a  join table of students and teachers, there might also be an attribute indicating which classroom they are in (though there migth be an argument for separating that to its own table as well).
+</br>
 12. Describe and diagram the relationship between patients and doctors.
 13. Describe and diagram the relationship between museums and original_paintings.
 14. What are some examples of acceptable values for the parts of an HTTP response?
+</br>
+
+* Header - 
+* Status code - 404
+* Body  - "Page not found"
+</br>
 15. What types of output do we want to test when we test our controllers?
+</br>
+Status codes?
```

Comment:
Yup! Status codes and responses. We didn't teach this info this time around, so well done!